Tests/SwiftFormatPrettyPrintTests/PrettyPrintTestCase.swift

```diff
@@ -6,37 +6,90 @@ import XCTest
 @testable import SwiftFormatPrettyPrint
 
 public class PrettyPrintTestCase: XCTestCase {
+  /// A helper that will keep track of which diagnostics have been emitted, and their locations.
+  private var consumer: DiagnosticTrackingConsumer? = nil
+
+  /// Implements `DiagnosticConsumer` to keep track which diagnostics have been raised and their
+  /// locations.
+  private class DiagnosticTrackingConsumer: DiagnosticConsumer {
+    var registeredDiagnostics = [(String, line: Int?, column: Int?)]()
```

Comment:
Dude, I was just about to make a similar change to the phase 1 rule tests so I could more accurately check the line/column numbers of the comment diagnostics I added. _Spooky_ 👻 

Later on we should merge things to have a common `XCTestCase` subclass for both rule "phases" that has this `XCTAssertDiagnosed` behavior in it.